bigtable_automation/java/dataflow/README.md

```diff
@@ -43,27 +56,69 @@ mvn compile
 If you have a one-off CSV file that you want to load into the test environment,
 run the following command:
 
-  ```
-  ./test.sh run-csv <BT_TABLE_NAME> <GCS_CSV_FILE>
-  ```
+```sh
+
+./test.sh run-csv <BT_TABLE_NAME> <GCS_CSV_FILE>
+
+```
 
 For example:
 
-  ```
-  ./test.sh run-csv mixer_test_bt gs://datcom-store-test/mixer_test_cache.csv
-  ```
+```sh
+
+./test.sh run-csv mixer_test_bt gs://datcom-store-test/mixer_test_cache.csv
+
+```
 
 Where, `mixer_test_bt` is the BT table that will be created from
 `gs://datcom-store-test/mixer_test_cache.csv`.
 
+### Flex Templates
+
+[Flex Templates](https://cloud.google.com/dataflow/docs/guides/templates/using-flex-templates) are a new generation of Dataflow templates that packages dependencies in the form of a Docker image instead of Jars.
+
+The same pipeline code can generate both Classic and Flex templates. Flex templates can be validated the same way using the steps above by adding "-flex" to the subcommands like the following.
+
+```sh
+./test.sh deploy-flex
+./test.sh run-cache-flex branch_2021_01_08_13_25_48
+./test.sh run-csv-flex mixer_test_bt gs://datcom-store-test/mixer_test_cache.csv
+```
+
 ## Production Deployment
 
 After validating the change in test environment, deploy it to PROD template by
 running the following command.
 
-```
+### Deploying Classic Template
+
+```sh
+
 mvn compile exec:java -Dexec.mainClass=org.datacommons.dataflow.CsvImport -Dexec.args="--runner=DataflowRunner --project=datcom-store --stagingLocation=gs://datcom-templates/staging --templateLocation=gs://datcom-templates/templates/csv_to_bt --region=us-central1 --usePublicIps=false"
+
 ```
 
 NOTE: Running this may throw an exception, but as long as it says `BUILD
 SUCCESS` the template has been updated.
+
+### Deploying Flex Template
+
+NOTE: If the version in pom.xml changes, please update TAG field below as well.
+
+```sh
+
+TAG="0.0.2-SNAPSHOT"
+
+mvn clean package && gcloud dataflow flex-template build \
```

Comment:
I'm comparing this to the above invocation.. where do DataflowRunner and usePublicIps get set?

Reply:
DataflowRunner
- Not needed because all Dataflow templates run on Dataflow Runners. I don't think it was ever needed in existing commands either.

usePublicIps
Missed this one. added "--disable-public-ips" (re-run again for testing, result below).

https://pantheon.corp.google.com/dataflow/jobs/us-central1/2022-12-16_23_54_35-12748127224196230376;mainTab=JOB_GRAPH;logsSeverity=ERROR;graphView=1?project=google.com:datcom-store-dev&pageState=(%22dfTime%22:(%22l%22:%22dfJobMaxTime%22))&mods=-monitoring_api_staging

Reply:
But --disable-public-ips isn't added to this command-line?  I see only in test.sh (which is used only for testing)